internal/app/api/handler/index/index.go

```diff
@@ -30,3 +38,69 @@ func (web *Handler) Index(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+func (web *Handler) AddFriend(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	var friend *types.Friend
+
+	if err := json.NewDecoder(r.Body).Decode(&friend); err != nil {
+		respond.Error(w, err, http.StatusBadRequest)
+		return
+	}
+	friend.Id = bson.NewObjectId()
```

Comment:
handler should know nothing about what kind of database to be used.

Reply:
Fixed